libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -53,6 +53,7 @@ void DagBlockPacketHandler::process(const PacketData &packet_data, const std::sh
     }
 
     if (auto status = syncing_handler_->checkDagBlockValidation(block); !status.first) {
+      trx_mgr_->insertBroadcastedTransactions(new_transactions);
```

Comment:
but then we do not need to send all trxs on `MissingHashes` in `GetDagSyncPacketsHandler` and also we should think about malicious peer and somehow later remove those trxs

Reply:
@MatusKysel  Lets put this into the issue